lib/schema/createModelsFromSchema.js

```diff
@@ -6,28 +6,74 @@ var Arrow = require('arrow'),
  */
 exports.createModelsFromSchema = function () {
 	var self = this,
-		models = {};
+		models = {},
+		exposePrimaryKeyAsId = _.has(Arrow.getGlobal(), 'config.serialization.exposePrimaryKeyAsId') ? _.get(Arrow.getGlobal(), 'config.serialization.exposePrimaryKeyAsId') : false;
 
-	Object.keys(self.schema.objects).forEach(function (modelName) {
-		var object = self.schema.objects[modelName],
-			fields = {};
-		Object.keys(object).forEach(function (fieldName) {
-			if (fieldName !== 'id') {
+	/**
+	 * Returns model's primary key column's name
+	 * @param {string} modelName
+	 * @returns {string}
+	 */
+	function getPrimaryKeyFromSchema(modelName) {
+		return self.schema.primary_keys[modelName];
+	}
+
+	/**
+	 * Get model fields except the primaryKey field
+	 * @param {string} modelName
+	 * @param {Object} modelSchema
+	 * @returns {Object}
+	 */
+	function getFieldsExceptPrimaryKey(modelName, modelSchema) {
+		return Object.keys(modelSchema)
+			.filter(function (fieldName) {
+				var primaryKeyColumnName = getPrimaryKeyFromSchema(modelName);
+				return exposePrimaryKeyAsId && typeof primaryKeyColumnName !== 'undefined' && fieldName === primaryKeyColumnName ? false : (fieldName === 'id' ? false : true);
```

Comment:
It should be enough to only filter fields that have the same name as the primary key column. No need for extra checks against the new `exposePrimaryKeyAsId` config as that only affects serialization.

On a side note: I prefer to split longer expressions like this into multiple lines with helper variables/functions for better readability. The ternary operator is awesome but it quickly gets a pain in the ass to read, especially with multiple comparisons and nested stuff.